src/dotdirdeps.cpp

```diff
@@ -108,6 +162,8 @@ void writeDotDirDepGraph(FTextStream &t,const DirDef *dd,bool linkRelations)
         //    shortName().data(),
         //    !usedDir->isParentOf(this)
         //    );
+        //! @todo consider adding used directories, which have a relation over more "generations"
+        //!       (grandparents, ...)
         if (dir!=usedDir && dir->parent()==usedDir->parent() && 
             !usedDir->isParentOf(dd))
           // include if both have the same parent (or no parent)
```

Comment:
This adds only used directories which have the same parent directory of the current directory or its parents. Thus there will not be a deep structure to the used directory.

For example

```
src/
├── a
│   └── aa
│       └── dependent.h
└── b
    └── bb
        └── dependency.h
```

In the directory dependency graph for `aa`, `aa` will only point to `b` as it is the first level of the common parent `src`. *Not* `bb`.